### ISSUE TITLE
Check for finiteness of all Model::f* outputs  

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/AMICI#514]
For a bunch of `Model` functions we currently check for nan and inf values. I would like to do that for all `Model::f*`to facilitate debugging.

I would guess that the runtime overhead is negligible, but we could also add a run-time or compile-time option in `Model` to enable/disable.

Any thoughts on that @FFroehlich?

**Comment by FFroehlich** [comment]
At the moment, the policy is to only apply checkFinite to those quantities that are touched by the solver. The runtime overhead of this can actually be single digit percentages of the overall runtime, so it is not completely negligible.

Still, I think adding more tests would be pretty valuable for debugging. At the same time, one could also add checks for negative states.

**Comment by dweindl** [comment]
Okay, so we add a run-time option to `Model`? Which defaults to on/off? I'd go for checking per default. Usually time required to reproduce a failure after enabling the option is much higher than the overhead.

**Comment by FFroehlich** [comment]
Hmm, I would rather have it off by default. Can we simplify reproducing the failure?

**Comment by dweindl** [comment]
> Hmm, I would rather have it off by default.

Also fine for me.

> Can we simplify reproducing the failure?

I thought mostly about cases where this happens only after a potentially large number of simulation, e.g. as part of optimization. Would have to provide a kind of "core dump"...

**Comment by FFroehlich** [comment]
Okay, lets open a new issue for that.
